app/formaters/DesFormatter.scala

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package formaters
+
+import java.text.DecimalFormat
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
+import javax.inject.{Inject, Singleton}
+import langswitch.LangMessages
+import model.des._
+import model.{PeriodKey, VrtRepaymentDetailData}
+import play.api.Logger
+import play.api.mvc.Request
+import req.RequestSupport
+
+@Singleton
+class DesFormatter @Inject() (addressFormater: AddressFormter, requestSupport: RequestSupport) {
+
+  import requestSupport._
+
+  def addMissingStatus(vrd: List[VrtRepaymentDetailData]): List[VrtRepaymentDetailData] = {
+
+    if (vrd.filter(f => f.repaymentDetailsData.riskingStatus == INITIAL.value).size > 0) {
+      vrd
+    } else {
+
+      val rdd: RepaymentDetailData = vrd(0).repaymentDetailsData.copy(riskingStatus          = INITIAL.value, lastUpdateReceivedDate = Some(vrd(0).repaymentDetailsData.returnCreationDate))
+      val vrtRepaymentDetailData: VrtRepaymentDetailData = vrd(0).copy(repaymentDetailsData = rdd)
+      vrtRepaymentDetailData :: vrd
+    }
+  }
+
+  def getReturnDebitChargeExists(financialData: Option[FinancialData], periodKey: PeriodKey): Boolean = {
+
+    val transactionIterable = for {
+      fdAllOption <- financialData
+    } yield (fdAllOption.financialTransactions.filter(f => (f.periodKey == periodKey.value && f.chargeType == "VAT Return Debit Charge")))
+
+    transactionIterable match {
```

Comment:
not necessarily better, but just another version of the same code 
 
`transactionIterable.exist(_.size > 0)` or `transactionIterable.exist(_.nonEmpty)`

;)